File: mutants/idi/devkit/create_agent_gui.py
#!/usr/bin/env python3
"""Agent Development GUI - Python version using tkinter.

Interactive GUI for creating and managing Tau Language intelligent agents.
Leverages Python's rapid prototyping and tkinter's simplicity.
"""

import tkinter as tk
from tkinter import ttk, filedialog, messagebox
from pathlib import Path
import json
import subprocess
import sys


class AgentDevGUI:
    def __init__(self, root):
        self.root = root
        self.root.title("IDI Agent Development GUI")
        self.root.geometry("600x500")
        
        # Variables
        self.agent_name = tk.StringVar()
        self.strategy = tk.StringVar(value="momentum")
        self.output_dir = tk.StringVar(value=str(Path("../practice").resolve()))
        
        self.setup_ui()
    
    def setup_ui(self):
        # Header
        header = ttk.Label(
            self.root,
            text="IDI Agent Development",
            font=("Arial", 16, "bold")
        )
        header.pack(pady=10)
        
        # Agent name
        frame_name = ttk.Frame(self.root)
        frame_name.pack(fill=tk.X, padx=20, pady=5)
        ttk.Label(frame_name, text="Agent Name:").pack(side=tk.LEFT)
        ttk.Entry(frame_name, textvariable=self.agent_name, width=30).pack(side=tk.LEFT, padx=10)
        
        # Strategy selection
        frame_strategy = ttk.Frame(self.root)
        frame_strategy.pack(fill=tk.X, padx=20, pady=5)
        ttk.Label(frame_strategy, text="Strategy:").pack(side=tk.LEFT)
        strategy_combo = ttk.Combobox(
            frame_strategy,
            textvariable=self.strategy,
            values=["momentum", "mean_reversion", "regime_aware"],
            state="readonly",
            width=27
        )
        strategy_combo.pack(side=tk.LEFT, padx=10)
        
        # Output directory
        frame_output = ttk.Frame(self.root)
        frame_output.pack(fill=tk.X, padx=20, pady=5)
        ttk.Label(frame_output, text="Output Dir:").pack(side=tk.LEFT)
        ttk.Entry(frame_output, textvariable=self.output_dir, width=25).pack(side=tk.LEFT, padx=10)
        ttk.Button(
            frame_output,
            text="Browse",
            command=self.browse_output_dir
        ).pack(side=tk.LEFT)
        
        # Strategy description
        self.desc_label = ttk.Label(
            self.root,
            text="Momentum following strategy",
            font=("Arial", 10, "italic"),
            foreground="gray"
        )
        self.desc_label.pack(pady=5)
        strategy_combo.bind("<<ComboboxSelected>>", self.on_strategy_change)
        
        # Buttons
        button_frame = ttk.Frame(self.root)
        button_frame.pack(pady=20)
        
        ttk.Button(
            button_frame,
            text="Create Agent",
            command=self.create_agent,
            width=20
        ).pack(side=tk.LEFT, padx=5)
        
        ttk.Button(
            button_frame,
            text="List Templates",
            command=self.list_templates,
            width=20
        ).pack(side=tk.LEFT, padx=5)
        
        # Status area
        self.status_text = tk.Text(self.root, height=10, wrap=tk.WORD)
        self.status_text.pack(fill=tk.BOTH, expand=True, padx=20, pady=10)
        scrollbar = ttk.Scrollbar(self.status_text)
        scrollbar.pack(side=tk.RIGHT, fill=tk.Y)
        self.status_text.config(yscrollcommand=scrollbar.set)
        scrollbar.config(command=self.status_text.yview)
        
        self.log("Ready. Enter agent name and click 'Create Agent'.")
    
    def on_strategy_change(self, event=None):
        descriptions = {
            "momentum": "Momentum following strategy",
            "mean_reversion": "Mean reversion strategy",
            "regime_aware": "Regime-aware adaptive strategy",
        }
        self.desc_label.config(text=descriptions.get(self.strategy.get(), ""))
    
    def browse_output_dir(self):
        directory = filedialog.askdirectory(initialdir=self.output_dir.get())
        if directory:
            self.output_dir.set(directory)
    
    def log(self, message):
        self.status_text.insert(tk.END, message + "\n")
        self.status_text.see(tk.END)
        self.root.update()
    
    def list_templates(self):
        try:
            # Call CLI to list templates
            result = subprocess.run(
                [sys.executable, "create_agent.py", "--list-templates"],
                capture_output=True,
                text=True,
                cwd=Path(__file__).parent
            )
            self.log(result.stdout)
            if result.stderr:
                self.log(f"Error: {result.stderr}")
        except Exception as e:
            self.log(f"Error listing templates: {e}")
    
    def create_agent(self):
        name = self.agent_name.get().strip()
        if not name:
            messagebox.showerror("Error", "Agent name is required")
            return
        
        strategy = self.strategy.get()
        output_dir = Path(self.output_dir.get())
        
        self.log(f"Creating agent '{name}' with strategy '{strategy}'...")
        self.log(f"Output directory: {output_dir}")
        
        try:
            # Call CLI to create agent
            result = subprocess.run(
                [
                    sys.executable,
                    "create_agent.py",
                    "--name", name,
                    "--strategy", strategy,
                    "--out", str(output_dir),
                ],
                capture_output=True,
                text=True,
                cwd=Path(__file__).parent
            )
            
            if result.returncode == 0:
                self.log(result.stdout)
                messagebox.showinfo("Success", f"Agent '{name}' created successfully!")
            else:
                self.log(f"Error: {result.stderr}")
                messagebox.showerror("Error", f"Failed to create agent:\n{result.stderr}")
        except Exception as e:
            self.log(f"Error: {e}")
            messagebox.showerror("Error", f"Failed to create agent: {e}")


def main():
    root = tk.Tk()
    app = AgentDevGUI(root)
    root.mainloop()


if __name__ == "__main__":
    main()

File: mutants/idi/devkit/rust/src/bin/create_agent_gui.rs
//! Agent Development GUI - Rust version using egui.
//!
//! Interactive GUI for creating and managing Tau Language intelligent agents.
//! Leverages Rust's performance and egui's immediate mode GUI.

use eframe::egui;
use std::path::PathBuf;

struct AgentDevApp {
    agent_name: String,
    selected_strategy: String,
    output_dir: PathBuf,
    status_message: String,
}

impl Default for AgentDevApp {
    fn default() -> Self {
        Self {
            agent_name: String::new(),
            selected_strategy: "momentum".to_string(),
            output_dir: PathBuf::from("../practice"),
            status_message: String::new(),
        }
    }
}

impl eframe::App for AgentDevApp {
    fn update(&mut self, ctx: &egui::Context, _frame: &mut eframe::Frame) {
        egui::CentralPanel::default().show(ctx, |ui| {
            ui.heading("IDI Agent Development GUI");
            ui.separator();
            
            ui.horizontal(|ui| {
                ui.label("Agent Name:");
                ui.text_edit_singleline(&mut self.agent_name);
            });
            
            ui.horizontal(|ui| {
                ui.label("Strategy:");
                egui::ComboBox::from_id_source("strategy")
                    .selected_text(&self.selected_strategy)
                    .show_ui(ui, |ui| {
                        ui.selectable_value(&mut self.selected_strategy, "momentum".to_string(), "Momentum");
                        ui.selectable_value(&mut self.selected_strategy, "mean_reversion".to_string(), "Mean Reversion");
                        ui.selectable_value(&mut self.selected_strategy, "regime_aware".to_string(), "Regime Aware");
                    });
            });
            
            ui.horizontal(|ui| {
                ui.label("Output Directory:");
                ui.text_edit_singleline(&mut self.output_dir.to_string_lossy().to_string());
            });
            
            ui.separator();
            
            if ui.button("Create Agent").clicked() {
                if self.agent_name.is_empty() {
                    self.status_message = "❌ Error: Agent name required".to_string();
                } else {
                    // TODO: Call create_agent_directory
                    self.status_message = format!("✅ Created agent '{}'", self.agent_name);
                }
            }
            
            if !self.status_message.is_empty() {
                ui.separator();
                ui.label(&self.status_message);
            }
        });
    }
}

fn main() -> Result<(), eframe::Error> {
    let options = eframe::NativeOptions {
        viewport: egui::ViewportBuilder::default()
            .with_inner_size([400.0, 300.0])
            .with_title("IDI Agent Development"),
        ..Default::default()
    };
    
    eframe::run_native(
        "IDI Agent Development",
        options,
        Box::new(|_cc| Box::new(AgentDevApp::default())),
    )
}

